### DISCUSSION TITLE
Disable autohiding of the Bottom / Sub Panels - Closed Tabs, Bookmarks, History

### DISCUSSION THREAD

**Discussion by tonybweb** [mbnuqw/sidebery#1919] [Q&A]
Is it possible to disable the autohiding of the bottom sub panels? I'd like to just keep them open and give them their own close button. I'm not opposed to some CSS hacking..

**Comment by soul4kills** [comment]
You can add bookmarks as one of your panels and it will stay open like a tab panel. It's in Sidebery Settings > Navigation Bar.

**Reply by tonybweb** [reply]
Yeah I'm aware, thanks for the suggestion . Unfortunately that's not exactly what I'm going for. I'd like my bookmarks to always be visible simultaneously below my normal tabs.

**Reply by soul4kills** [reply]
Fo sho. I took a quick peek and made it stay open. If you're gonna hack it. It's not a simple one to do. It's not worth the effort because the tab panel gets disabled, so you can't scroll, can't interact with tabs. You would have to hack a whole bunch of elements individually.

Good luck. Hope you figure it out.

**Comment by tonybweb** [comment]
I also tinkered with firefox's built-in tools and came to the conclusion it would need source code changes on the javascript side. CSS alone won't do it. I might dive deeper and give it a go someday, for now, it's a bit much. 

Oh well, this is still an amazing extension..

**Reply by soul4kills** [reply]
You don't need JavaScript. You can definitely do it with css. You'll have to use a lot of [attributes*=""] to override the inline stuff. You'll have to put it in userContent.css. Don't give up.

For example in this code is to modify the tab preview that i hacked together. Learned a lot trying to figure this out.
```
div.popup[style*="--fg"][style*="--bg"] {
	visibility: collapse;
		&:has(div[style*="background-image: url("][style*="opacity: 1;"]) {
		visibility: visible !important;
		border: 1px solid var(--tab-selected-outline-color);
		border-radius: 50%;
		box-shadow: 0 1px 20px -5px var(--tab-selected-outline-color)!important;
	}
	.header { visibility: collapse !important; }
	@media not (-moz-bool-pref: "sidebar.position_start") { left: auto !important;  right: calc(var(--uc-sidebar-hover-width) - var(--uc-sidebar-width)) !important;}
	@media (-moz-bool-pref: "sidebar.position_start") { right: auto !important; left: calc(var(--uc-sidebar-hover-width) - var(--uc-sidebar-width)) !important; }
}
	/* Tab preview fix alignment when uc.chromeless is toggled to true */
@media (-moz-bool-pref: "uc.chromeless") {
	div.popup[style*="--fg"][style*="--bg"] {
		@media not (-moz-bool-pref: "sidebar.position_start") { left: auto !important; right: var(--uc-sidebar-hover-width) !important;}
		@media (-moz-bool-pref: "sidebar.position_start") { right: auto !important; left: var(--uc-sidebar-hover-width) !important;}
	}
}
	/* Fix pop-in-page tab preview alignment - Specifically when maximized or uc.full.tabs is true */
@media (width: 1748px), (-moz-bool-pref: "uc.full.tabs") { /* Adjustments  needed for user specific screen size */
	div.popup[style*="--fg"][style*="--bg"]{
		@media not (-moz-bool-pref: "sidebar.position_start") { left: auto !important; right: 0px !important;}
		@media (-moz-bool-pref: "sidebar.position_start") { right: auto !important; left: 0px !important;}
	}
}
#sdbr_preview_root {
	@media not (-moz-bool-pref: "sidebar.position_start") { left: auto !important; right: 0px !important; }
	@media (-moz-bool-pref: "sidebar.position_start") { right: auto !important; left: 0px !important; }
}
```

So basically, I modified it to be able to check what side your sidebar is. And it automatically adjusts the offset depending on what width I have my sidebar at. It overrides any settings applied in Sidebery Settings. It was fun figuring it out.

Edit: it also gets rid of the annoying blank previews over unloaded tabs.

**Reply by soul4kills** [reply]
Here's a video of it.

https://github.com/user-attachments/assets/c7ba1987-126b-413b-b7d2-6623604e3662
